### ISSUE TITLE
Semantic segmentation branch is not getting trained

### ISSUE THREAD

**Issue by chandrakantkhandelwal** [DavyNeven/fastSceneUnderstanding#6]
I have trained enet on my dataset and it's giving good results. I used this enet model to initialise and train fastscene network on the same dataset for learning instances of car(I have dropped the depth branch).  

I am facing following problems while training the fastscene model:
1. The semantic segmentation results gets bad as the training progresses and never improves afterwards(I understand the dip in accuracy initially, as the initialised weights are tuned to semantic segmentation, and as training progresses they changes. However I think it should improve as the training moves forward)
2. The instance segmentation results are just ok at training time, at test time as they depends on semantic segmentation branch output the results automatically becomes bad.  

Note: The training set size is around 6k images (height:480, width:640, channels:3) and none of the classes were heavily skewed in train set. I tried training with learning rate 1e-3, 5e-4, 1e-4, 5e-5, none of them worked. I used enet style class-weights, the batch size was 6 and used adam optimizer.

I even tried using the released fastscene model weights on cityscapes to initialise the network and train it on my dataset, however this too didn't help. Please advise what else could I try to train it on any other dataset then cityscapes.

**Comment by dbbert** [comment]
We managed to train the published model successfully on the [CVPPP plant leaf segmentation dataset](https://www.plant-phenotyping.org/CVPPP2017-challenge) and on an internal highway driving dataset with only small tweaks to the hyperparameters. 

Perhaps you can try to tune the balancing parameter between the semantic and instance loss (putting more emphasis on the semantic loss), or start out debugging by only training the semantic segmentation branch to see if it is a capacity problem. If that is the case you could try using in a larger resnet-style network rather than ENet, or increase capacity by adding some more layers or using more filters per layer.

Good luck!

**Comment by chandrakantkhandelwal** [comment]
Thanks @dbbert for the quick response. Sure I will try to play with the balancing parameter, if it doesn't works will move to resnet style instead of adding more layers to the current enet network.
